### DISCUSSION TITLE
IKEv1 Reauth time calculation

### DISCUSSION THREAD

**Discussion by mianatag** [strongswan/strongswan#1657] [General]
In my setup, the reauthentication in IKEv1 is leading to traffic loss. Couple of questions related to this:

1. How is the reauth time for IKEv1 calculated. 
2. Is there a way to control the reauth time in IKEv1? I am using ipsec.conf file in my setup so would need recommendations on the parameters to be used in this.
3. Are the rekey time and reauth time calculations dependent on one another?
4. https://docs.strongswan.org/docs/5.9/config/rekeying.html mentions "Since the rekeying of an SA needs some time, the margin values must not be set too low." How low can this value be set?
5. Does reauth_time in swanctl.conf even control ikev1 reauth time?

**Comment by tobiasbrunner** [comment]
> 1. How is the reauth time for IKEv1 calculated.

Refer to the page you already saw. Although, with ipsec.conf the formula is basically `rekeytime = lifetime - (margintime + random(0, margintime * rekeyfuzz))`.

> 2\. Is there a way to control the reauth time in IKEv1?

Unless the reauthentication is initiated by the peer, sure. By the way, you shouldn't be using IKEv1 anymore.

> I am using ipsec.conf file in my setup so would need recommendations on the parameters to be used in this.

You can try to translate the parameters via [this page](https://wiki.strongswan.org/projects/strongswan/wiki/Fromipsecconf).

> 3\. Are the rekey time and reauth time calculations dependent on one another?

There is no IKE rekeying with IKEv1, so it doesn't matter which option is used (in swanctl.conf, in ipsec.conf the `reauth` option is irrelevant for IKEv1).

> 4\. "Since the rekeying of an SA needs some time, the margin values must not be set too low." How low can this value be set?

Depends on [retransmission settings](https://docs.strongswan.org/docs/5.9/config/retransmission.html) etc. Just apply common sense.

> 5\. Does reauth_time in swanctl.conf even control ikev1 reauth time?

Sure, it just has the same effect as `rekey_time`. Configuring both doesn't really make sense with IKEv1, though.
